### ISSUE TITLE
Flake? Creating containers for a running step not present on any worker

### ISSUE THREAD

**Issue by topherbullock** [concourse/concourse#2637]
From @xtreme-sameer-vohra 's comment on #2588

> We just observed an issue where a job had a step that was in progress for ~24hrs without any output or updates.
> The DB stated that the container for the step was in `creating` mode. The worker had no state for the container.
> The ATC had been re-started due to another issue ~24hrs ago, which corresponds with the time since when the container's state went wonky.
> 

Maybe there's an edge case in https://github.com/concourse/concourse/blob/master/atc/worker/container_provider.go#L87 or the code which calls it in the `exec/*_step` where the container in the DB won't ever be created for a running step if the ATC is restarted?

**Comment by jama-pivotal** [comment]
Some steps to try for reproducing this:
- "Just look at it"; take a look at the code and see if it might get stuck somewhere
- Try adding some stuff in the database to replicate this case where the container exists in the DB as creating but it hasn't been created
- Add some sleeps?

**Comment by topherbullock** [comment]
Picking this up, and I'm going to : 
![](https://media.giphy.com/media/xT1XH0jQoN1UnHybeM/giphy.gif)

**Comment by topherbullock** [comment]
That was **way** too easy. I think I've managed to stick my test pipeline of 10 `echo "hello world"` tasks using the same container image. Triggered 30 builds and it was trucking along okay, then I chucked the worker process into `/sys/fs/cgroup/freezer/worker/` and froze it a few times ( simulating a poor worker connection, since its forwarded, and just general slowness / resiliency  )

Going to see if it stays stuck overnight.
